### PR TITLE
Add support for failureDomain to MD and MP

### PR DESCRIPTION
**Pull request by primeroz**
Towards https://github.com/giantswarm/giantswarm/issues/25361

### CODE REVIEW

**Comment by primeroz** [comment]
```
data:
  values: |
    metadata:
      name: "cluster02"
      description: "cluster02 test cluster"
      organization: "multi-project"

    providerSpecific:
      location: "westeurope"
      #This is the GHOST subscription
      subscriptionId: 6b1f6e4a-6d0e-4aa4-9a5a-fbaca65a23b3

    controlPlane:
      replicas: 1

    nodePools:
    - name: md10
      instanceType: Standard_D2s_v5
      replicas: 2
      rootVolumeSizeGB: 50
      failureDomain: "1"
    - name: md20
      instanceType: Standard_D2s_v5
      replicas: 2
      rootVolumeSizeGB: 50
      failureDomain: "2"

kind: ConfigMap
metadata:
  name: cluster02-user-values
  namespace: org-multi-project

```

```
➜ kubectl topology node
NAME                                     REGION       ZONE
cluster02-control-plane-87c46025-zzwbv   westeurope   westeurope-1
cluster02-md10-efb0145f-m4jcc            westeurope   westeurope-1
cluster02-md10-efb0145f-zxwvf            westeurope   westeurope-1
cluster02-md20-efb0145f-2qbpq            westeurope   westeurope-2
cluster02-md20-efb0145f-d25j7            westeurope   westeurope-2
```

File: helm/cluster-azure/values.schema.json [hunk 1]
```diff
@@ -333,6 +333,15 @@
                         "title": "Disable HealthChecks for the MachineDeployment",
                         "type": "boolean"
                     },
+                    "failureDomain": {
+                        "enum": [
+                            "1",
+                            "2",
+                            "3"
```
Comment (bavarianbidi) [inline helm/cluster-azure/values.schema.json:340]: Is every Azure region consist of three failureDomains? Or can there be Azure regions with only one of them?

**Comment by primeroz** [comment]
as discussed in planning i will change the `default` to be the same as now

**Review by bavarianbidi — APPROVED** [review]

**Comment by primeroz** [comment]
final test 

* Creating a cluster with current code gets us same as now 
```
    nodePools:
    - name: md00
      instanceType: Standard_D2s_v5
      replicas: 3
      rootVolumeSizeGB: 50
```
* Cluster topology
```
➜ kubectl topology node
NAME                                     REGION       ZONE
cluster02-control-plane-87c46025-htvkx   westeurope   westeurope-2
cluster02-md00-efb0145f-gqxs8            westeurope   0
cluster02-md00-efb0145f-qpv9c            westeurope   0
cluster02-md00-efb0145f-v4fqv            westeurope   0
```

* Creating a cluster with 
```
    nodePools:
    - name: md10
      instanceType: Standard_D2s_v5
      replicas: 3
      rootVolumeSizeGB: 50
      faillureDomain: 1
    - name: md20
      instanceType: Standard_D2s_v5
      replicas: 3
      rootVolumeSizeGB: 50
      faillureDomain: 2
```
* Gets us 
```

```

**Pull request merged**